### PR TITLE
Move Https LightyServerBuilder to Lighty-jetty-server module

### PR DESCRIPTION
**Pull request by PeterSuna**
Move files belonging to ligty-jetty-server: RncRestConfConfigUtils,
https server builder and server config files to ligty-jetty-server module.

Divide jetty-server configuration from restconf configuration.

Rebased on: #1025
ID: 1819

### CODE REVIEW

File: lighty-modules/lighty-jetty-server/src/main/java/io/lighty/server/util/LightyServerConfigUtils.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+        } catch (IOException | NoSuchAlgorithmException | CertificateException | KeyStoreException e) {
+            throw new ConfigurationException("Unable to create KeyStore configuration", e);
+        }
+    }
+
+    private static Optional<InputStream> readKeyStoreFile(final String keyStoreFilePath) throws IOException {
+        InputStream ksFile;
+        try {
+            LOG.info("Trying to load KeyStore from filesystem from path {}", keyStoreFilePath);
+            ksFile = Files.newInputStream(Paths.get(keyStoreFilePath));
```
[the remark this answers is not in the document]
Comment (PeterSuna) [reply lighty-modules/lighty-jetty-server/src/main/java/io/lighty/server/util/LightyServerConfigUtils.java:108]: ignore

File: lighty-modules/lighty-jetty-server/pom.xml [hunk 2]
```diff
@@ -22,6 +22,10 @@
     <url>https://github.com/PANTHEONtech/lighty</url>
 
     <dependencies>
+        <dependency>
```
[the remark this answers is not in the document]
Comment (PeterSuna) [reply lighty-modules/lighty-jetty-server/pom.xml:25]: ignore

**Review by ihrasko — CHANGES_REQUESTED** [review]

File: lighty-applications/lighty-rnc-app-aggregator/lighty-rnc-module/src/main/java/io/lighty/applications/rnc/module/config/RncLightyModuleConfigUtils.java [hunk 3]
```diff
@@ -34,26 +36,28 @@ private RncLightyModuleConfigUtils() {
...
-            restconfConfig = new RncRestConfConfiguration(RncRestConfConfigUtils
-                    .getRestConfConfiguration(Files.newInputStream(configPath)));
-            restconfConfig.setSecurityConfig(RncRestConfConfigUtils.createSecurityConfig(restconfConfig));
+            restConfConfiguration = RestConfConfigUtils
+                    .getRestConfConfiguration(Files.newInputStream(configPath));
+            LOG.debug("lighty.io RESTCONF module configuration from file loaded!");
+
+            LOG.debug("Loading lighty.io Jetty server module configuration from file...");
+            lightyServerConfig = LightyServerConfigUtils.getServerConfiguration(Files.newInputStream(configPath));
             LOG.debug("lighty.io RESTCONF module configuration from file loaded!");
```
[the remark this answers is not in the document]
Comment (PeterSuna) [reply lighty-applications/lighty-rnc-app-aggregator/lighty-rnc-module/src/main/java/io/lighty/applications/rnc/module/config/RncLightyModuleConfigUtils.java]: Done

File: lighty-applications/lighty-rnc-app-aggregator/lighty-rnc-module/src/main/java/io/lighty/applications/rnc/module/config/RncLightyModuleConfigUtils.java [hunk 4]
```diff
@@ -66,47 +70,48 @@ public static RncLightyModuleConfiguration loadConfigFromFile(Path configPath) t
...
         LOG.debug("Loading default lighty.io RESTCONF module configuration...");
-        RncRestConfConfiguration restconfConfig =
-                new RncRestConfConfiguration(RncRestConfConfigUtils.getDefaultRestConfConfiguration());
-        restconfConfig.setSecurityConfig(RncRestConfConfigUtils.createSecurityConfig(restconfConfig));
+        final RestConfConfiguration restConfConfiguration = RestConfConfigUtils.getDefaultRestConfConfiguration();
+        LOG.debug("Default lighty.io RESTCONF module configuration loaded!");
+
+        LOG.debug("Loading default lighty.io Jetty server module configuration...");
+        final LightyServerConfig lightyServerConfig = LightyServerConfigUtils.getDefaultLightyServerConfig();
         LOG.debug("Default lighty.io RESTCONF module configuration loaded!");
```
[the remark this answers is not in the document]
Comment (PeterSuna) [reply lighty-applications/lighty-rnc-app-aggregator/lighty-rnc-module/src/main/java/io/lighty/applications/rnc/module/config/RncLightyModuleConfigUtils.java]: Done

File: lighty-modules/lighty-jetty-server/pom.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (PeterSuna) [reply lighty-modules/lighty-jetty-server/pom.xml:25]: ignore

File: lighty-applications/lighty-rnc-app-aggregator/lighty-rnc-module/src/main/java/io/lighty/applications/rnc/module/RncLightyModule.java [hunk 5]
```diff
@@ -94,56 +96,55 @@ public boolean initModules() {
...
-        if (confConf.isUseHttps()) {
-            this.jettyServerBuilder = new HttpsLightyServerBuilder(inetSocketAddress, config.getSecurityConfig());
+        if (serverConfig.isUseHttps()) {
+            this.jettyServerBuilder = new HttpsLightyServerBuilder(inetSocketAddress, serverConfig.getSecurityConfig());
         } else {
             this.jettyServerBuilder = new LightyServerBuilder(inetSocketAddress);
         }
 
-        return CommunityRestConfBuilder.from(confConf)
+        return CommunityRestConfBuilder.from(restConfConfiguration)
```
Comment (ihrasko) [inline lighty-applications/lighty-rnc-app-aggregator/lighty-rnc-module/src/main/java/io/lighty/applications/rnc/module/RncLightyModule.java:131]: What about?

```
    private CommunityRestConf initRestconf(final RestConfConfiguration config, final LightyServerConfig serverConfig) {
        final InetSocketAddress inetSocketAddress =
                new InetSocketAddress(config.getInetAddress(), config.getHttpPort());

        if (serverConfig.isUseHttps()) {
            this.jettyServerBuilder = new HttpsLightyServerBuilder(inetSocketAddress, serverConfig.getSecurityConfig());
        } else {
            this.jettyServerBuilder = new LightyServerBuilder(inetSocketAddress);
        }

        return CommunityRestConfBuilder.from(config)
            .withLightyServer(jettyServerBuilder)
            .build();
    }
```

**Review by ihrasko — CHANGES_REQUESTED** [review]

File: lighty-applications/lighty-rnc-app-aggregator/lighty-rnc-module/src/main/java/io/lighty/applications/rnc/module/RncLightyModule.java (hunk 5 above)
Comment (PeterSuna) [reply lighty-applications/lighty-rnc-app-aggregator/lighty-rnc-module/src/main/java/io/lighty/applications/rnc/module/RncLightyModule.java:131]: If you mean by this to remove from code:
```java
final RestConfConfiguration restConfConfiguration = RestConfConfigUtils.getRestConfConfiguration(config, services);
```

Then this is how lighty services are copied to Restconf configuration.
https://github.com/PANTHEONtech/lighty/blob/master/lighty-modules/lighty-restconf-nb-community/src/main/java/io/lighty/modules/northbound/restconf/community/impl/util/RestConfConfigUtils.java#L134

https://github.com/PANTHEONtech/lighty/blob/48c87f70707d06d03dc2f1c16af7468902bd1012/lighty-examples/lighty-community-restconf-actions-app/src/main/java/io/lighty/examples/controllers/actions/Main.java#L154

Comment (ihrasko) [reply lighty-applications/lighty-rnc-app-aggregator/lighty-rnc-module/src/main/java/io/lighty/applications/rnc/module/RncLightyModule.java:131]: OK

**Review by ihrasko — APPROVED** [review]

**Pull request merged**